python/pyspark/java_gateway.py

```diff
@@ -33,14 +33,18 @@
 from pyspark.serializers import read_int
 
 
-def launch_gateway(conf=None):
+def launch_gateway(conf=None, popen_kwargs=None):
     """
     launch jvm gateway
     :param conf: spark configuration passed to spark-submit
+    :param popen_kwargs: Dictionary of kwargs to pass to Popen when spawning
```

Comment:
Mention that this is a developer feature and may change in future versions.

Reply:
And ... you said that right here. Doh! I'll update the docstring at least.